bgpd/bgp_conditional_adv.c

```diff
@@ -119,6 +120,7 @@ static void bgp_conditional_adv_routes(struct peer *peer, afi_t afi,
 			RESET_FLAG(dummy_attr.rmap_change_flags);
 
 			ret = route_map_apply(rmap, dest_p, &path);
+			advmap_attr = bgp_attr_intern(&dummy_attr);
```

Comment:
Shouldn't this be uninterned somewhere if `subgroup_announce_check` returns `false`?

Reply:
Or if routemap returns `DENY`? There are a lot of cases where we need to do the unintern. Probably the code should be reworked in some way and we intern the attr only when it's necessary.

Reply:
Added unintern stuff around.

Reply:
I'll update a bit a current code.

Reply:
@idryzhov updated.